include/opal.h

```diff
@@ -96,6 +96,23 @@ typedef enum lexeme_type
   lx_Input
 } lexeme_type_e;
 
+/// Struct for keyword string/type
+typedef struct keyword
+{
+  char *str;
+  lexeme_type_e lex_type;
+} keyword;
```

Comment:
The `char *str` member could be `const char *str` since this struct should only read it & never write to it.